delta/plugins/search/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/search/SearchConfigUpdaterSuite.scala

```diff
@@ -0,0 +1,114 @@
+  private val defaultEsProjection   = esProjection.copy(
+    id = defaultProjectionId,
+    query = indexingConfig.query,
+    mapping = indexingConfig.mapping,
+    indexGroup = searchGroup,
+    context = indexingConfig.context,
+    settings = indexingConfig.settings,
+    resourceTypes = indexingConfig.resourceTypes
+  )
+  private val modifiedEsProjection  = defaultEsProjection.copy(resourceTypes = Set(iri"https://schema.org/Person"))
+  private val compositeValue        = CompositeViewValue(
+    Some(defaults.name),
+    Some(defaults.description),
+    NonEmptySet.of(projectSource.copy(id = defaultSourceId)),
+    NonEmptySet.of(defaultEsProjection),
+    None
+  )
+  private val modifiedCompositeView = compositeValue.copy(projections = NonEmptySet.of(modifiedEsProjection))
+
+  private val proj1 = ProjectRef.unsafe("org", "proj1")
+  private val proj2 = ProjectRef.unsafe("org", "proj2")
+  private val proj3 = ProjectRef.unsafe("org", "proj3")
+  private val rev   = 1
+
+  private val upToDateView        = ActiveViewDef(ViewRef(proj1, defaultViewId), uuid, rev, compositeValue)
+  private val defaultOutdatedView = ActiveViewDef(ViewRef(proj2, defaultViewId), uuid, rev, modifiedCompositeView)
+  private val customOutdatedView  = ActiveViewDef(ViewRef(proj2, id), uuid, rev, modifiedCompositeView)
+  private val deprecatedView      = DeprecatedViewDef(ViewRef(proj3, defaultViewId))
+
+  private val views        =
+    Stream.emits(Seq(upToDateView, defaultOutdatedView, customOutdatedView, deprecatedView)).zipWithIndex.map {
+      case (v, index) =>
+        SuccessElem(
+          tpe = CompositeViews.entityType,
+          id = v.ref.viewId,
+          project = Some(v.ref.project),
+          instant = Instant.EPOCH,
+          offset = Offset.at(index),
+          value = v,
+          rev = 1
+        )
+    }
+  private val updatedViews = MutableSet.empty[(CompositeViewDef, CompositeViewFields)]
+  implicit private class CompositeViewTestOps(x: (CompositeViewDef, CompositeViewFields)) {
+    def viewDef: CompositeViewDef = x._1
+  }
+
+  private val update: (ActiveViewDef, CompositeViewFields) => Task[Unit] =
+    (viewDef, fields) => Task { updatedViews.add((viewDef, fields)) }.void
+
+  test("Update the views") {
+    new SearchConfigUpdater(defaults, indexingConfig)(views, update).compile.drain
+  }
+
+  test("A default active view whose sources do not match the current config should be updated") {
+    assert(updatedViews.contains((defaultOutdatedView, fromValue(upToDateView.value))))
+  }
+
+  test("A default active view whose sources match the current config should not be updated") {
+    assert(!updatedViews.exists(_.viewDef == upToDateView))
+  }
+
+  test("A non-default active view should not be updated") {
+    assert(!updatedViews.exists(_.viewDef == customOutdatedView))
+  }
+
+  test("Deprecated view should not be updated") {
+    assert(!updatedViews.exists(_.viewDef == deprecatedView))
+  }
+
```

Comment:
It would also be nicer to have each of these test running individually, rather than having to all be ran in order